receiver/sqlqueryreceiver/config.go

```diff
@@ -28,16 +31,126 @@ type Config struct {
 	Queries                                 []Query `mapstructure:"queries"`
 }
 
+func (c Config) Validate() error {
+	if c.Driver == "" {
+		return errors.New("'driver' cannot be empty")
+	}
+	if c.DataSource == "" {
+		return errors.New("'datasource' cannot be empty")
+	}
+	if len(c.Queries) == 0 {
+		return errors.New("'queries' cannot be empty")
+	}
+	for _, query := range c.Queries {
+		if err := query.Validate(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 type Query struct {
-	SQL     string   `mapstructure:"sql"`
-	Metrics []Metric `mapstructure:"metrics"`
+	SQL     string      `mapstructure:"sql"`
+	Metrics []MetricCfg `mapstructure:"metrics"`
 }
 
-type Metric struct {
-	MetricName       string   `mapstructure:"metric_name"`
-	ValueColumn      string   `mapstructure:"value_column"`
-	AttributeColumns []string `mapstructure:"attribute_columns"`
-	IsMonotonic      bool     `mapstructure:"is_monotonic"`
+func (q Query) Validate() error {
+	var errs error
+	if q.SQL == "" {
+		errs = multierr.Append(errs, errors.New("'query.sql' cannot be empty"))
+	}
+	if len(q.Metrics) == 0 {
+		errs = multierr.Append(errs, errors.New("'query.metrics' cannot be empty"))
+	}
+	for _, metric := range q.Metrics {
+		if err := metric.Validate(); err != nil {
+			errs = multierr.Append(errs, err)
+		}
+	}
+	return errs
+}
+
+type MetricCfg struct {
+	MetricName       string            `mapstructure:"metric_name"`
+	ValueColumn      string            `mapstructure:"value_column"`
+	AttributeColumns []string          `mapstructure:"attribute_columns"`
+	Monotonic        bool              `mapstructure:"monotonic"`
+	ValueType        MetricValueType   `mapstructure:"value_type"`
+	DataType         MetricDataType    `mapstructure:"data_type"`
+	Aggregation      MetricAggregation `mapstructure:"aggregation"`
+	Unit             string            `mapstructure:"unit"`
+	Description      string            `mapstructure:"description"`
+}
```

Comment:
Do we want to consider adding an optional tags map that allow users to set custom tags on these metrics? I suppose they could always use a processor to do it but would be an easy add and would be nice to have the option.

Reply:
Hi @Konig-Corey, I think this could be done in a separate PR if other receivers are doing it as well. Otherwise I like the idea of letting a processor do it 🙂.

Reply:
@pmcollins  I submitted a PR on your branch if we think we want to add it. I think allowing someone to set them explicitly on the receiver makes sense instead of having to intercept these metrics in a processor to add tags. An example use case could be to add a tag(s) for table name, db instance etc.

Reply:
Are other receivers doing this as well? If so, can you submit the PR against the main branch after this one gets merged?

Reply:
@pmcollins no problem. Anyway we can expedite this getting merged in? It seems this PR has been open for quite some time.

Reply:
@dmitryax appears to be appending approval from you.